src/main/java/TaskList.java

```diff
@@ -0,0 +1,133 @@
+import java.util.ArrayList;
+
+public class TaskList {
+
+    private ArrayList<Task> taskList = new ArrayList<Task>();
+
+    /**
+     * gets details for all available task when the list command is invoked.
+     */
+    public String getTaskList() {
+        String message = "Here are the tasks in your list: \n";
+        String tasks = "";
+
+        for (int i = 0; i < taskList.size(); i++) {
+            Task task = taskList.get(i);
+            int count = i + 1;
+            tasks = tasks + count + "." + task.toString() + "\n";
+
+        }
+        return message + tasks;
+    }
+
+    /**
+     * returns details for all available task when the list command is invoked.
+     */
+    public String getTasksData() {
+        String message = "";
+        for (int i = 0; i < taskList.size(); i++) {
+            Task task = taskList.get(i);
+            int count = i + 1;
+            message = message + task.toString() + "\r\n";
+
+        }
+        return message;
+    }
+
+    /**
+     * marks a task as done and prints message that a task is marked as done.
+     * returns message associated with a task marked as completed.
+     *
+     * @param taskIndex position of a task in the taskList Arraylist
+     *
+     */
+
+    public String markDone(int taskIndex) {
+        String msg;
+        try {
+            Task t = taskList.get(taskIndex);
+            t.setCompleted();
+            msg = "Nice! I've marked this task as done: \n";
+            msg  = msg + "   [✓] " + t.getTask();
+
+        } catch (IndexOutOfBoundsException ex) {
+            if (taskList.size() == 0) {
+                msg = "No tasks to be done.";
+            } else {
+                taskIndex += 1;
+                msg = "Task " + taskIndex + " does not exist.";
+            }
+        }
+        return msg;
+    }
+
+    /**
+     * deletes a task from taskList Arraylist and prints message that a task is deleted.
+     * returns a string which indicates a task has been deleted.
+     *
+     * @param taskIndex position of a task in the taskList Arraylist
+     *
+     */
+    public String deleteTask(int taskIndex) {
+        String msg = "";
```

Comment:
Consider using a StringBuilder instead.

Reply:
Also should decouple the string results from TaskList. You should have a DeleteCommand that takes care of that.